core/shared/src/main/scala/org/http4s/Header.scala

```diff
@@ -170,20 +170,28 @@ object Header {
         h: H
     )(implicit H: Header[H, _]): Header.ToRaw with Primitive =
       new Header.ToRaw with Primitive {
-        val values = List(Header.Raw(H.name, H.value(h)))
+        val values = Header.Raw(H.name, H.value(h)) :: Nil
       }
 
     implicit def foldablesToRaw[F[_]: Foldable, H](
         h: F[H]
     )(implicit convert: H => ToRaw with Primitive): Header.ToRaw = new Header.ToRaw {
-      val values: List[Raw] = h.toList.foldMap(v => convert(v).values)
+      val values = h
+        .foldLeft(collection.mutable.ListBuffer.empty[Header.Raw]) { (buf, v) =>
```

Comment:
Instead of using a `ListBuffer` shouldn't we use a `List.newBuilder`? I'm not an expert on internal collections implementations, but I believe `List.newBuilder` has an optimization to avoid re-building the list in `toList`. I'm less confident `ListBuffer` has that same optimization because it is a collection of its own.

Reply:
@armanbilge they should be the same:
```
scala> val lb = List.newBuilder
val lb: scala.collection.mutable.Builder[Nothing, List[Nothing]] = ListBuffer()
                                                                                                                                                                                                        
scala> lb.getClass.getName
val res0: String = scala.collection.mutable.ListBuffer
```
However, I would use `List.newBuilder` as well, just because it does not require neither importing nor specifying a path to it.

Reply:
Ah, so it is! TIL, thanks :)

https://github.com/scala/scala/blob/98a42345385cc8adccde36c21f3192d4b4862087/src/library/scala/collection/immutable/List.scala#L686